src/dotnet/Chat.UI.Blazor/Components/Settings/NightModeToggle.razor

```diff
@@ -0,0 +1,20 @@
+@inherits ComputedStateComponent<bool>
+
+<Toggle Value="@_isDarkTheme" ValueChanged="OnThemeChanged" Class="toggle-sm"/>
+
+@code {
+    private bool _isDarkTheme;
+
+    [Inject] ThemeUI ThemeUI { get; init; } = null!;
+
+    private void OnThemeChanged()
+        => ThemeUI.CurrentTheme.Value = ThemeUI.CurrentTheme.ValueOrDefault == Theme.Dark ? Theme.Light : Theme.Dark;
```

Comment:
Can you clarify this trick?

Reply:
I mean that value from input is not used - you are always trying to invert value stored in ThemeUI